src/functions/template-tags/general.php

```diff
@@ -1380,11 +1380,15 @@ function tribe_events_get_the_excerpt( $post = null, $allowed_html = null ) {
 			$excerpt = wp_trim_words( $excerpt, $excerpt_length, $excerpt_more );
 		}
 
-		// Remove all shortcode Content before removing HTML
-		if ( ! $allow_shortcode ) {
-			$excerpt = preg_replace( '#\[.+\]#U', '', $excerpt );
+		// If shortcode filter is enabled lets process them
+		if ( $allow_shortcode ) {
+			$excerpt = do_shortcode( $excerpt );
 		}
 
```

Comment:
...misread your latest change: that is indeed what you're doing so we're all good here. Optionally, we could add a filter to control whether cleanup (`preg_replace( '#\[.+\]#U', '', $excerpt )`) takes place or not.